oaknational/Oak-Web-Application
# Implement subject header

## Description

- Implemented an updated `CurriculumHeaderTabNav` component for the subject header
- Refreshed the subject header styling
- Updated several components to support the new CurriculumHeaderTabNav
- Added a new `flat` button variant
- Modified CurriculumHeaderTabNav stories and tests to account for the updated tabular navigation.
- Visually removed redundant Curriculum tab H2 headings, so that they are now only visible to screen readers
    - A change was also made in the [oak-components](https://github.com/oaknational/oak-components/pull/298) repo to the `OakDownloadsJourneyChildSubjectTierSelector` component to facilitate this 
- @rightsaidjames please note that it was discussed with @HelenHarlow that for the 'selected tab' hover over should have no effect – so if in the course of your testing you're hovering over the currently selected tab and nothing occurs that is by design. Mouse hover over should however affect the remaining two unselected tabs however.

## Issue(s)

Fixes #CUR-869

## How to test

1. Go to https://deploy-preview-2852--oak-web-application.netlify.thenational.academy.
2. Navigate to the Teachers' Curriculum section.
3. Select a subject and phase.
4. You should see the new `CurriculumHeaderTabNav` component with the new navigation tabs

## Screenshots

How it used to look (delete if n/a):

<img width="726" alt="image" src="https://github.com/user-attachments/assets/b3b9d255-054d-495b-a34b-8967db61e8cf">

How it should now look:

<img width="1458" alt="image" src="https://github.com/user-attachments/assets/931ac7c2-9921-420a-ab8c-d0972618512c">

## Checklist

- [x] Manually tested on Chrome
- [ ] Does this PR update a package with a breaking change – No
